lib/connection/connect.js

```diff
@@ -157,4 +253,39 @@ function makeConnection(family, options, callback) {
   socket.once('connect', connectHandler);
 }
 
+const CONNECTION_ERROR_EVENTS = ['error', 'close', 'timeout', 'parseError'];
+function runCommand(conn, ns, command, options, callback) {
```

Comment:
I expect some discussion around this method. Ideally our wire protocol primitives (`command`, `getMore`, etc) would be placed directly on the `Connection` class, in which case I could just use `conn.command(...)`. Unfortunately, that's not the case today. If there is consensus, I could move _just_ `command` onto `Connection`, since this will be the same method that we use for authentication.

Reply:
I'd prefer to leave `command` in the wire protocol until we can move all of the wire protocol primitives to `Connection`. I think it would be less organized to only move `command` here.

Reply:
Agree with @kvwalker: lets not move these over piecemeal.